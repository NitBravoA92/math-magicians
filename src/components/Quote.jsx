import useQuote from '../hooks/useQuote';

const Quote = () => {
  const { loading, data, error } = useQuote({});
  console.log('loading: ' + loading, 'data: ' + data?.quote, 'error: ' + error);
  return (
    <section id="quotes" className="page-content">
      <div className="container">
        <div id="display-quote">
          <p id='quote-text' className={`${JSON.stringify(data) === '{}' ? 'd-none' : ''}`}>
            <i>{`"${data.quote}",`}</i><b>{`${data.author}`}</b>
          </p>
          <p id='loading-message' className={`${!loading ? 'd-none' : ''}`}>{ loading ? 'Loading Quote...' : '' }</p>
          <p id='error-message' className={`${!error ? 'd-none' : ''}`}>{ error || '' }</p>
        </div>
      </div>
    </section>
  )
}

export default Quote;