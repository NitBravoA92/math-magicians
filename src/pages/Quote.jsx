import Layout from './Layout';
import QuoteItem from '../components/QuoteItem';

const Quote = () => (
  <Layout>
    <section id='quote'>
      <div className='page-content'>
        <QuoteItem />
      </div>
    </section>
  </Layout>
);

export default Quote;