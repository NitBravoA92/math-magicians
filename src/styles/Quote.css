#quote-text {
  font-size: 1.5em;
  text-align: center;
  line-height: 1.2;
}

.d-none {
  display: none;
}

#quote {
  font-style: italic;
  color: #121111;
}

#author {
  color: #f50057;
}

#loading-message,
#error-message {
  margin-top: 1.5rem;
  text-align: center;
  line-height: 1.2;
}

#loading-message {
  font-size: 4em;
  color: #d1d1d1;
}

#error-message {
  font-size: 3em;
  color: #ba1c00;
}

@media screen and (min-width: 768px) {
  #quote-text {
    font-size: 2.3em;
  }
}
