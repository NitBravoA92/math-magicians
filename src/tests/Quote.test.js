import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Quote from '../pages/Quote';

describe('The Quote component', () => {
  test('renders correctly into the DOM', () => {
    const component = renderer.create(<Quote />).toJSON();

    expect(component).toMatchSnapshot();
  });

  test("display the message 'Loading Quote...'", () => {
    const component = render(<Quote />);

    expect(component.container).toHaveTextContent('Loading Quote...');
  });

  test('should render the QuoteItem component', () => {
    const component = render(<Quote />);

    const quoteContainer = component.container.querySelector('#display-quote');

    expect(quoteContainer).toBeInTheDocument();
  });

  test("shouldn't display an error message in the first rendered", () => {
    const component = render(<Quote />);

    const errorMessageContainer = component.container.querySelector('#error-message');

    expect(errorMessageContainer).toHaveClass('d-none');
  });
});
