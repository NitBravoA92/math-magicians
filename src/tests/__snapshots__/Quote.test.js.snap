// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Quote component renders correctly into the DOM 1`] = `
Array [
  <header>
    <a
      className="logo-container"
      href="/"
    >
      <img
        alt="logo"
        className="App-logo"
        src="mathLogo.gif"
      />
      <span
        className="logo-title"
      >
        Math Magicians
      </span>
    </a>
    <nav>
      <ul
        id="menu"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className={null}
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className={null}
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main>
    <section
      id="quote"
    >
      <div
        className="page-content"
      >
        <div
          id="display-quote"
        >
          <p
            className="d-none"
            id="quote-text"
          >
            <span
              id="quote"
            />
            <span
              id="author"
            >
               â€” 
            </span>
          </p>
          <p
            className=""
            id="loading-message"
          >
            Loading Quote...
          </p>
          <p
            className="d-none"
            id="error-message"
          />
        </div>
      </div>
    </section>
  </main>,
]
`;
